### PR TITLE
Parse Commit Messages in Pull Requests for Issue IDs

### PR DESCRIPTION
**Pull request by brwarner**
PR to parse commit messages of commits within a Pull Request when deciding which JIRA issues to associate with the PR.

It does this by sending a REST API call for the commit list every time a pull_request webhook comes in. The messages are added to the string that's parsed for Jira issue ids. I've also added the `pull_request.synchronize` event to the list of subscribed webhooks to capture new commits being added to a PR.

### Rationale

According to Jira's Documentation, one can associate a pull request with an issue by doing at least one of the following:

* Include a commit in the pull request that has the issue key in the commit message. Note, the commit cannot be a merge commit.
* Include the issue key in the pull request title.
* Ensure that the source branch name includes the issue key.

Source: https://support.atlassian.com/jira-software-cloud/docs/reference-issues-in-your-development-work/

GitHub is listed as one of the supported integrations, but bullet (1) is not satisfied. Right now only the title of a pull request is parsed for issue mentions.

### CODE REVIEW

**Comment by brwarner** [comment]
I've added a transform unit test for the new behavior and testing the build locally on a developer Jira instance. It was successfully able to pick up issues mentioned only in commits on the PR but not within the title itself.

Unfortunately, I can't verify the non-transform unit tests locally. Despite getting the project building and running a bunch of unit tests (under the `safe` folder and others) were failing for me on a fresh repo. I want to make sure they're not broken by watching the GitHub workflow run before I tried adding my own tests.

The error I seem to get is that the `app.receive` in the `safe\pull-request.test.js` tests doesn't seem to do anything, and as a resultthe `td.verify` calls fail. I tried stepping through with my debugger but it never makes it into any of the webevent handlers.

**Comment by brwarner** [comment]
Also while this does not close https://github.com/integrations/jira/issues/119, scanning commit messages is mentioned with just as much importance in the comments.

**Comment by mboudreau** [comment]
Thanks for your contribution, that's great :)

We'll get back to you soon, but to me, this sounds like a slam dunk.

Cheers!

**Comment by brwarner** [comment]
I wrapped the commit retrieval webhook in a try/catch similar to the new reviewers code in the same file for consistency.

I did notice in my local tests that now only the branch and commit fields are getting set in Jira, but not the PR field. It was working before, so I'm not sure if any of the other changes merged into the file broke something. I'll do more investigating soon.

File: lib/github/index.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ module.exports = (robot) => {
     'pull_request.reopened',
     'pull_request.edited',
     'pull_request_review',
+    'pull_request.synchronize',
```
Comment (mboudreau) [inline lib/github/index.js:34]: can you please move this one line up so we can group all `pull_request.X` events together?

**Review by mboudreau — CHANGES_REQUESTED** [review]
Very small change.  As part of this, we also need to update the Github app webhooks we're listening for.  Not sure if we should wait for the ownership transfer of the integration to be done first before deploying this.  Will have to confer with the team/Github.

**Comment by AmrutaKawade** [comment]
> The error I seem to get is that the `app.receive` in the `safe\pull-request.test.js` tests doesn't seem to do anything, and as a resultthe `td.verify` calls fail. I tried stepping through with my debugger but it never makes it into any of the webevent handlers.

How did you fix this? Even I am getting similar error

**Comment by brwarner** [comment]
> > The error I seem to get is that the `app.receive` in the `safe\pull-request.test.js` tests doesn't seem to do anything, and as a resultthe `td.verify` calls fail. I tried stepping through with my debugger but it never makes it into any of the webevent handlers.
> 
> How did you fix this? Even I am getting similar error

Hey,

As I mentioned, I never fixed it because I was getting the error even on a fresh repository so it didn't seem to have anything to do with my code. I also didn't know how to fix it.

**Comment by mboudreau** [comment]
@AmrutaKawade @brwarner You can ignore those tests.  There's been a major refactor done to typescript which showed _a lot_ of issues in the tests and other parts of the code which have now been fixed.  You can view these changes in the `micros` branch which we are getting ready for the migration from Github ownership to Atlassian ownership.  

We haven't forgotten about this PR, we just need to complete the migration first before we can start pumping out features, fixing issues and merging PRs.  So, please hang tight :)

**Comment by asford** [comment]
@mboudreau This feature looks amazing, is there another issue or PR we can track for status to determine when this will be unblocked?

**Comment by brwarner** [comment]
Hey, now that you guys have merged in your `micros` branch and your contribution steps are _drastically_ improved, I think I'll remake this PR soon atop the new codebase unless you already have something in the works. 

Should be far easier to test this time around.

**Comment by mboudreau** [comment]
@brwarner that would be amazing, thank you!

It should be much easier to add now.  I'm happy to assist with any questions you may have or if you want to pair on this problem over video chat.  It would also be great for you to give feedback on the contributors documentation - if anything is missing or not clear enough.

Cheers!

**Review by jFJ13 — APPROVED** [review]

**Comment by mboudreau** [comment]
@brwarner Should I just close this PR if you're just going to make a new one?

**Comment by brwarner** [comment]
Yup. See you again soon, Atlassian friends.

**Pull request closed without merging**